addons/bus/static/src/services/multi_tab_service.js

```diff
@@ -0,0 +1,213 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { browser } from '@web/core/browser/browser';
+import session from 'web.session';
+
+/**
+ * This class uses a Master/Slaves with Leader Election architecture in
+ * order to keep track of the main tab. Tabs are synchronized thanks to the
+ * localStorage.
+ *
+ * localStorage used keys are:
+ * - {LOCAL_STORAGE_PREFIX}.{sanitizedOrigin}.lastPresenceByTab:
+ *   mapping of tab ids to their last recorded presence.
+ * - {LOCAL_STORAGE_PREFIX}.{sanitizedOrigin}.main : id of the current
+ *   main tab.
+ * - {LOCAL_STORAGE_PREFIX}.{sanitizedOrigin}.heartbeat : last main tab
+ *   heartbeat time.
+ *
+ * trigger on env.bus:
+ * - become_main_tab : when this tab became the main.
+ * - no_longer_main_tab : when this tab is no longer the main.
+ */
+export class MultiTab {
+    constructor(env) {
+        this.env = env;
+
+        // CONSTANTS
+        this.TAB_HEARTBEAT_PERIOD = 10000; // 10 seconds
+        this.MAIN_TAB_HEARTBEAT_PERIOD = 1500; // 1.5 seconds
+        this.HEARTBEAT_OUT_OF_DATE_PERIOD = 5000; // 5 seconds
+        this.HEARTBEAT_KILL_OLD_PERIOD = 15000; // 15 seconds
+        this.LOCAL_STORAGE_PREFIX = 'multiTabService';
+
+        // PROPERTIES
+        this._isOnMainTab = false;
+        this._sanitizedOrigin = session.origin.replace(/:\/{0,2}/g, '_');
+
+        const now = new Date().getTime();
+        this._id = _.uniqueId(this.LOCAL_STORAGE_PREFIX) + ':' + now;
+        browser.addEventListener('unload', this._onUnload.bind(this));
+        browser.addEventListener('storage', this._onStorage.bind(this));
+        // REGISTER THIS TAB
+        const lastPresenceByTab = this._callLocalStorage('getItem', 'lastPresenceByTab', {});
+        lastPresenceByTab[this._id] = now;
+        this._callLocalStorage('setItem', 'lastPresenceByTab', lastPresenceByTab);
+
+        if (!this._callLocalStorage('getItem', 'main')) {
+            this._startElection();
+        }
+        this._heartbeat();
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    isOnMainTab() {
```

Comment:
also, I think that the multi_tab service should expose a way to broadcast a message on all other tabs. This is actually a very useful method